streams/src/main/java/org/apache/kafka/streams/Topology.java

```diff
@@ -304,6 +304,7 @@ public synchronized Topology addSource(final String name,
      * @return itself
      * @throws TopologyException if processor is already added or if topics have already been registered by name
      */
+    @SuppressWarnings("overloads")
```

Comment:
Why do we need this? Does the build report a WARNING and fails without it? (same below)

Reply:
Adding default implementation to close() and configure() make Deserializer be a functional interface. Hence, it is feasible to pass lambdas without explicit type (Deserializer and TimestampExtractor can be the identical lambda expression). javac doesn't like the potential ambiguity. We can reorder the arguments in order to "avoid" the complaint. However, Topology is a public class so changing the method signature will break the compatibility. As a result, I added SuppressWarnings.

Reply:
Thanks for the details -- however, you did not answer my question: Does the build fail without the annotation?

Reply:
> Does the build fail without the annotation?

absolutely :)